packages/backend/src/payment-method/ilp/spsp/middleware.ts

```diff
@@ -38,7 +39,7 @@ const spspMiddleware = async (
   ctx: SPSPWalletAddressContext,
   next: () => Promise<unknown>
 ): Promise<void> => {
-  if (ctx.accepts('application/spsp4+json')) {
+  if (ctx.request.header.accept?.includes(SPSP_CONTENT_TYPE_V4)) {
```

Comment:
this `ctx.accepts` was the reason why by default when fetching the wallet address, the SPSP details were returned, since usually, the `accept` header included catch all `*/*` route, which to `ctx.accepts` would be truthy. 

Now, we need to explicitly add `application/spsp4+json` in the header to get the SPSP details